fastvm/libevmjit/Compiler.cpp

```diff
@@ -566,13 +566,19 @@ void Compiler::compileBasicBlock(BasicBlock& _basicBlock, RuntimeManager& _runti
 		case Instruction::ANY_DUP:
 		{
 			auto index = static_cast<size_t>(inst) - static_cast<size_t>(Instruction::DUP1);
+			if ((m_rev < EVM_AION_V1) && (index > 16)) {
+				goto invalidInstruction;
+			}
 			stack.dup(index);
 			break;
 		}
 
 		case Instruction::ANY_SWAP:
 		{
 			auto index = static_cast<size_t>(inst) - static_cast<size_t>(Instruction::SWAP1) + 1;
+			if ((m_rev < EVM_AION_V1) && (index > 16)) {
+				goto invalidInstruction;
+			}
```

Comment:
I think this is problematic. `0x90` is `SWAP1` but interpreted as `DUP17` for old versions. Also, did you update `ANY_DUP` and `ANY_SWAP`?

Reply:
I was wondering whether it would help if we use unallocated byte code rather than replacing old ones.

Reply:
yeah, replacing original instructions needs more concern. That's a lesson from this issue. reassign  DUP17 = 0xb0, SWAP17 = 0xc0